tests/integration/base.py

```diff
@@ -373,6 +374,38 @@ def generate_and_settle(self, wait_interfaces=None, state_dir=None):
 
         if 'Run \'systemctl daemon-reload\' to reload units.' in out:
             print('\nWARNING: systemd units changed without reload:', out)
+
+        self.settle(wait_interfaces, state_dir)
+
+    def try_and_settle(self, try_func, wait_interfaces_try=None, wait_interfaces_after=None):
+        '''Run `netplan try`, wait for the configuration to apply, run `try_func` and allow try to time out'''
+
+        # Because `netplan try` uses select(2) to implement the timeout, stdin
+        # needs to be a stream that is never ready to read.
+        fifo_path = f"/tmp/netplan-try-test.{random.randint(10000, 99999)}"
+        os.mkfifo(fifo_path)
+        fifo = os.open(fifo_path, os.O_RDONLY | os.O_NONBLOCK)
+
+        proc = subprocess.Popen(["netplan", "try", "--timeout=5"], stdin=fifo)
+
+        # wait for apply to finish
+        while not pathlib.Path("/run/netplan/netplan-try.ready").exists():
+            time.sleep(0.1)
```

Comment:
I wonder if timeouts are big enough here, especially for slow architectures, like riscv64 or loong64. In other tests (`regressions.py:test_try_accept/reject_*`) we're mostly working with 5 second delays, before triggering any action. So you should consider increasing that `--timeout=5` to something slightly bigger.

Furthermore, we might want to add an explicit termination condition for the `while` loop, to an avoid endless loop when `netplan try --timeout=X` does not work correctly or create the stamp in time. E.g. count the iterations and explicitly break after `TIMEOUT*10` iterations of `sleep(0.1)`.